Jenkinsfile

```diff
@@ -0,0 +1,158 @@
+library 'magic-butler-catalogue'
+
+def PROJECT_NAME = "eslint-plugin-logdna"
+def REPO = "logdna/${PROJECT_NAME}"
+def TRIGGER_PATTERN = ".*@logdnabot.*"
+
+pipeline {
+  agent none
+
+  options {
+    timestamps()
+    ansiColor 'xterm'
+  }
+
+  triggers {
+    issueCommentTrigger(TRIGGER_PATTERN)
+  }
+
+  stages {
+    stage('Validate PR Source') {
+      when {
+        expression { env.CHANGE_FORK }
+        not {
+          triggeredBy 'issueCommentCause'
+        }
+      }
+      steps {
+        error("A maintainer needs to approve this PR for CI by commenting")
+      }
+    }
```

Comment:
I wonder if its possible to include all of this in a function in the magic butter repo. A function that returns a stage.

Reply:
that would be ideal, there's a few step definitions that are identical across all of our jenkinsfiles.